src/odemis/gui/cont/streams.py

```diff
@@ -544,6 +548,28 @@ def _on_metadata_btn(self, evt):
                         v = u"[%s … %s]" % (u", ".join(str(a) for a in v[:20]), u", ".join(str(a) for a in v[-20:]))
                     text += u"%s: %s\n" % (key, v)
 
+        # only display extra settings once
+        if model.MD_EXTRA_SETTINGS in raw[0].metadata:
+            text += u"\n======================================\nHardware Settings\n"
+            text += u"======================================\n"
+            for comp, vas in md[model.MD_EXTRA_SETTINGS].items():
+                if vas:
+                    text += u"Component %s:\n" % comp
+                for name, content in vas.items():
```

Comment:
Please put this entire block in a try/except so that if the metadata is not the expected format it will just skip showing the component, and move on.